src/PluginsHelper.php

```diff
@@ -131,4 +139,281 @@ public static function get_plugin_data( $plugin ) {
+		$results           = array();
+		$errors            = new \WP_Error();
+
+		foreach ( $plugins as $plugin ) {
+			$slug = sanitize_key( $plugin );
+
+			if ( isset( $existing_plugins[ $slug ] ) ) {
+				$installed_plugins[] = $plugin;
+				continue;
+			}
+
+			$api = plugins_api(
+				'plugin_information',
+				array(
+					'slug'   => $slug,
+					'fields' => array(
+						'sections' => false,
+					),
+				)
+			);
+
+			if ( is_wp_error( $api ) ) {
+				$properties = array(
+					/* translators: %s: plugin slug (example: woocommerce-services) */
+					'error_message' => __( 'The requested plugin `%s` could not be installed. Plugin API call failed.', 'woocommerce-admin' ),
+					'api'           => $api,
+					'slug'          => $slug,
+				);
+				wc_admin_record_tracks_event( 'install_plugin_error', $properties );
+
+				do_action( 'woocommerce_plugins_install_api_error', $slug, $api );
+
+				$errors->add(
+					$plugin,
+					sprintf(
+						/* translators: %s: plugin slug (example: woocommerce-services) */
+						__( 'The requested plugin `%s` could not be installed. Plugin API call failed.', 'woocommerce-admin' ),
+						$slug
+					)
+				);
+
+				continue;
+			}
+
+			$upgrader           = new \Plugin_Upgrader( new \Automatic_Upgrader_Skin() );
+			$result             = $upgrader->install( $api->download_link );
+			$results[ $plugin ] = $result;
+
+			if ( is_wp_error( $result ) || is_null( $result ) ) {
+				$properties = array(
+					/* translators: %s: plugin slug (example: woocommerce-services) */
+					'error_message' => __( 'The requested plugin `%s` could not be installed.', 'woocommerce-admin' ),
+					'slug'          => $slug,
+					'api'           => $api,
+					'upgrader'      => $upgrader,
+					'result'        => $result,
+				);
+				wc_admin_record_tracks_event( 'install_plugin_error', $properties );
+
+				do_action( 'woocommerce_plugins_install_error', $slug, $api, $result, $upgrader );
+
+				$errors->add(
+					$plugin,
+					sprintf(
+						/* translators: %s: plugin slug (example: woocommerce-services) */
+						__( 'The requested plugin `%s` could not be installed. Upgrader install failed.', 'woocommerce-admin' ),
+						$slug
+					)
+				);
+				continue;
+			}
+
+			$installed_plugins[] = $plugin;
+		}
+
+		$data = array(
+			'installed' => $installed_plugins,
+			'results'   => $results,
+			'errors'    => $errors,
+		);
+
+		if ( $job_id ) {
+			self::set_installation_status( $job_id, 'complete', $data );
+		}
+
+		return $data;
+	}
+
+	/**
+	 * Schedule plugin installation.
+	 *
+	 * @param array $plugins Plugins to install.
+	 * @return string Job ID.
+	 */
+	public static function schedule_install_plugins( $plugins ) {
+		if ( empty( $plugins ) || ! is_array( $plugins ) ) {
+			return new \WP_Error( 'woocommerce_plugins_invalid_plugins', __( 'Plugins must be a non-empty array.', 'woocommerce-admin' ), 404 );
+		}
+
+		$job_id = uniqid();
+		self::set_installation_status( $job_id, 'pending' );
```

Comment:
Would it be worth including the list of `$plugins` here as part of the data.
`$data = array( 'pending' => $plugins )`?
In theory we would keep track the job id and the plugin information on the client side, if a user does a full refresh of the browser we would loose which plugins are pending.
This would also be helpful incase we trigger a job for each individual plugin.

Reply:
I think that's a great idea!  Will plan on adding pending AS/transient decision.